src/duration.js

```diff
@@ -432,6 +457,57 @@ export default class Duration {
     if (s === "P") s += "T0S";
     return s;
   }
+  
+  /**
+   * Returns an ISO 8601-compliant string representation of this Duration, formatted as a time of day.
+   * Note that this will return null if the duration is either negative, or equal to or greater than 24 hours.
+   * @see https://en.wikipedia.org/wiki/ISO_8601#Times
+   * @param {Object} opts - options
+   * @param {string} [opts.format='extended'] - choose between the basic and extended format
+   * @param {boolean} [opts.suppressPrefix='auto'] - choose between suppressing the `T` prefix auto, always or never
+   * @param {boolean} [opts.suppressMilliseconds=true] - exclude milliseconds from the format if they're 0
+   * @param {boolean} [opts.suppressSeconds=true] - exclude seconds from the format if they're 0
+   * @param {boolean} [opts.suppressMinutes=false] - exclude minutes from the format if they're 0
+   * @example Duration.fromObject({ hours: 20, milliseconds: 6 }).toISOTime() //=> '20:00:00.006'
+   * @example Duration.fromObject({ hours: 20 }).toISOTime() //=> '20:00'
+   * @example Duration.fromObject({ hours: 20 }).toISOTime({ format: 'basic' }) //=> 'T2000'
+   * @example Duration.fromObject({ hours: 20 }).toISOTime({ suppressPrefix: 'never' }) //=> 'T20:00'
+   * @example Duration.fromObject({ hours: 20 }).toISOTime({ suppressSeconds: false }) //=> '20:00:00'
+   * @example Duration.fromObject({ hours: 20 }).toISOTime({ suppressMinutes: true }) //=> 'T20'
+   * @return {string}
+   */
+  toISOTime(opts = {}) {
+    if (!this.isValid) return null;
+
+    const millis = this.toMillis();
+    if (millis < 0 || millis >= 86400000) return null;
+    
+    opts = Object.assign({
+      format: "extended",
+      suppressPrefix: "auto",
+      suppressMilliseconds: true,
+      suppressSeconds: true,
+      suppressMinutes: false
+    }, opts);
+
+    const value = this.shiftTo("hours", "minutes", "seconds", "milliseconds");
+    
+    let format =
+      value.milliseconds > 0 || !opts.suppressMilliseconds ? "hh:mm:ss.SSS" :
+      value.seconds > 0 || !opts.suppressSeconds ? "hh:mm:ss" :
+      value.minutes > 0 || !opts.suppressMinutes ? "hh:mm" :
+      "hh";
+
+    if (opts.format !== "extended") {
+      format = format.replace(/:/g, "");
+    }
+
+    if (opts.suppressPrefix === "never" || opts.suppressPrefix === "auto" && (format === "hh" || format === "hhmm")) {
```

Comment:
I'm not against options like these, but we need to make sure they're consistent. We don't support these elsewhere, do we?

Reply:
Yep, `DateTime` has similar options, so I designed this to behave the same way.
See https://github.com/moment/luxon/blob/master/src/datetime.js#L1534

Reply:
Specifically regarding the prefix suppression - this is unique to the time format:

> As of ISO 8601-1:2019, the basic format is T[hh][mm][ss] and the extended format is T[hh]:[mm]:[ss]. Earlier versions omitted the T in both formats.
> ...
> ISO 8601-1:2019 allows the T to be omitted in the extended format, as in "13:47:30", but only allows the T to be omitted in the basic format when there is no risk of ambiguity with date expressions.
>
> https://en.wikipedia.org/wiki/ISO_8601#Times

So I made the default behavior here `auto`, i.e. omit if possible - but if you're interacting with a system that always/never expects the prefix, this option allows you to override that behavior.